test/jdk/java/awt/image/ConvolveOp/KernelInitialisationTest.java

```diff
@@ -25,36 +25,49 @@
  * @test
  * @bug     8368729
  * @summary Tests that passing invalid values to Kernel constructor
- *          throws only IllegalArgumentException
+ *          throws only IllegalArgumentException or NullPointerException
  */
 
 import java.awt.image.Kernel;
 
 public class KernelInitialisationTest {
-    private static void expectIllegalArgumentException(Runnable code) {
+
+    private static void test(int width, int height, float[] data,
+                             Class<?> expected)
+    {
+        System.out.printf("Testing for width: %d, height: %d, data: %s%n",
+                          width, height, data == null ? "null" : "not null");
+        Class<?> actual = null;
         try {
-            code.run();
-            throw new RuntimeException("Expected IllegalArgumentException" +
-                " but no exception was thrown");
-        } catch (IllegalArgumentException e) {
-            // we expect IllegalArgumentException
+            new Kernel(width, height, data);
+        } catch (Exception e) {
+            actual = e.getClass();
+        }
+        if (actual != expected) {
+            System.err.println("Expected: " + expected);
+            System.err.println("Actual: " + actual);
+            throw new RuntimeException("Test failed");
```

Comment:
Does it make sense to include the class names in the error message too? It helps analysing the failure, you know the reason without opening the log file.